src/core/shape_inference/include/slice_shape_inference_utils.hpp

```diff
@@ -134,7 +134,11 @@ inline int64_t get_sliced_value(const int64_t& dim, const int64_t& start, const
     constexpr int64_t inf_bound = -1;
 
     const auto& norm_dim = dim == inf_bound ? std::numeric_limits<int64_t>::max() : dim;
+#if INTPTR_MAX == INT64_MAX
```

Comment:
instead of custom macro, we can use OPENVINO_ARCH_64_BIT / OPENVINO_ARCH_32_BIT:
https://github.com/openvinotoolkit/openvino/blob/1028c7b5d5479cb58216d88b6dd3dec1bb45ea33/src/core/include/openvino/core/visibility.hpp#L48-L66